common/include/pcl/PolygonMesh.h

```diff
@@ -23,6 +24,74 @@ namespace pcl
 
     std::vector< ::pcl::Vertices>  polygons;
 
+    /** \brief Inplace concatenate two pcl::PolygonMesh
+      * \param[in,out] mesh1 the first input and output mesh
+      * \param[in] mesh2 the second input mesh
+      * \return true if successful, false otherwise (unexpected error)
+      */
+    static bool
+    concatenate (pcl::PolygonMesh &mesh1, const pcl::PolygonMesh &mesh2)
+    {
+      // Make the resultant polygon mesh take the newest stamp
+      mesh1.header.stamp = std::max(mesh1.header.stamp, mesh2.header.stamp);
+
+      const auto point_offset = mesh1.cloud.width * mesh1.cloud.height;
+      std::transform(mesh2.polygons.begin (),
+                     mesh2.polygons.end (),
+                     std::back_inserter (mesh1.polygons),
+                     [point_offset](const auto& polygon)
+                     {
+                        auto poly = polygon;
+                        std::for_each(poly.vertices.begin (),
+                                      poly.vertices.end (),
+                                    [point_offset](auto& point_idx)
+                                    {
+                                      return point_idx + point_offset;
+                                    });
+                        return poly;
+                      });
+      mesh1.polygons.insert(mesh1.polygons.end(), mesh2.polygons.begin(), mesh2.polygons.end());
+
+      mesh1.cloud += mesh2.cloud;
```

Comment:
This operation may fail if the clouds are not compatible. Therefore, I think we should:
 1. Move this to the beginning of the function;
 2. Use `concatenate()` function and check it's return value.